### PR TITLE
FE-301: Cracking The Case With Flexbox ; FE-302: The CSS Grid Layout

### PR DESCRIPTION
**Pull request by nicoRDL**
## Description
- implements additional flex box styling to sample site
- updates site structure to fit flex box styling

- added improvements to use of Global and local grids

## Related Ticket
https://codeandtheory.atlassian.net/browse/FE-652
https://codeandtheory.atlassian.net/browse/FE-653

## Testing Instructions
1. Pull on branch FE-301/FE-652
2. Inspect and evaluate use of flex boxes
3. Inspect and evaluate use of Grid

## Risks
N/A

### CODE REVIEW

File: css/home-page.css [hunk 1]
```diff
@@ -152,7 +153,7 @@
   padding: 1rem;
 }
 
-@media screen and (min-width: 768px) {
+@media screen and (min-width: 48rem) {
```
Comment (lycoangelo) [inline css/home-page.css]: Let's retain the `px` unit for media queries.

File: css/home-page.css [hunk 2]
```diff
@@ -195,7 +198,7 @@
   }
 }
 
-@media screen and (min-width: 1024px) {
+@media screen and (min-width: 64rem) {
```
Comment (lycoangelo) [inline css/home-page.css]: Let's retain the `px` unit for media queries.

**Review by lycoangelo — CHANGES_REQUESTED** [review]

**Comment by nicoRDL** [comment]
@lycoangelo added requested changes d9551481e295d2a0ff465fd746ef11f35768535b

d9551481e295d2a0ff465fd746ef11f35768535b

File: view/index.html [hunk 3]
```diff
@@ -1,104 +1,174 @@
...
-                </a>
-                <a class="category-container featured-product-section" href="https://www.gloriousgaming.com/products/glorious-gmmk-pro-75-barebone-black" target="_blank">
-                  <h2 class="category-header">SPOTLIGHT</h2>
-                </a>
-              </div>
-              
+      <section class="categories-section">
+        <header id="main-title" class="homepage-header">
+          <h1>CLACKS & THOCC</h1>
+        </header>
```
Comment (lycoangelo) [inline view/index.html]: We can remove the `header` here and use only the `h1` tag. `h1` is already a heading tag and the `header` tag doesn't add any value to the semantic structure here.

File: view/index.html [hunk 4]
```diff
@@ -1,104 +1,174 @@
...
+        <div class="categories-container">
+          <a
+            class="category-container upper-categories feature-section"
+            href="article-page.html"
+          >
+            <h2 class="category-header">FEATURED</h2>
+          </a>
```
Comment (lycoangelo) [inline view/index.html:53]: The same here. We can remove the `h2` tag and use only the `tag`. for proper use of header tags, this link is a good read: https://blog.hubspot.com/marketing/header-tags#:~:text=What%20are%20header%20tags%3F,and%20SEO%20of%20a%20webpage.

File: view/index.html [hunk 5]
```diff
@@ -1,104 +1,174 @@
...
-                <h2>Down-sizing your keyboard: should you even?</h2>
+        </div>
+      </section>
+      <section
+        class="articles-container homepage-articles-section"
+        id="articles-section"
+      >
+        <span class="articles-container-header homepage-articles-header">
+          == Check out the Newest Articles ==
+        </span>
```
Comment (lycoangelo) [inline view/index.html]: I just checked the page design layout, and this element should be a header tag. Visually, it serves as the header of the Newest Articles.

File: view/index.html [hunk 6]
```diff
@@ -1,104 +1,174 @@
...
-                Author is the primary contributor to the site, both content and structure. Author is growing enthusiast for custom builds and modding.
-              </p>
-            </div>
-        </section>
+        </div>
+        <a href="browse-page.html">
+          <button class="articles-section-expand-btn">
+            == See More Articles ==
+          </button>
+        </a>
```
Comment (lycoangelo) [inline view/index.html]: [`a`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) and [`button`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) tags should not be a child of any other clickable elements such as `a` and `button` tags.

**Review by lycoangelo — CHANGES_REQUESTED** [review]
Hi @nicoRDL, I left a few comments. Please apply it to the rest of the other implementation.

**Comment by nicoRDL** [comment]
@lycoangelo added PR change requests and suggestions.
d08d1a0a9cb8b04da7dac8f88f2186731e3ee22a

File: view/browse-page.html [hunk 7]
```diff
@@ -82,179 +82,199 @@ <h2>Typing adjustment and how to make it easier</h2>
         <h1 class="articles-container-header">
           == Continue exploring below ==
         </h1>
```
Comment (lycoangelo) [inline view/browse-page.html]: There should be only 1 `h1` per page. Let's change this to `h2` instead.

File: view/browse-page.html [hunk 8]
```diff
@@ -82,179 +82,199 @@ <h2>Typing adjustment and how to make it easier</h2>
...
-            eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
-          <button class="article-card-expand-btn">
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+            <button class="article-card-expand-btn">
+              <a href="detail-page-light-sample1.html">continue reading ...</a>
+            </button>
```
Comment (lycoangelo) [inline view/browse-page.html]: @nicoRDL, there's still plenty of `a` tag inside `button` instances. Please check all the HTML files.

File: view/browse-page.html [hunk 9]
```diff
@@ -82,179 +82,199 @@ <h2>Typing adjustment and how to make it easier</h2>
...
+            <img
+              class="article-card-img"
+              src="../images/featureImage-Typing.jpeg"
+              alt="compact keyboard"
+            />
+          </figure>
+          <div class="article-card-text">
             <a href="detail-page-light-sample1.html">
-              continue reading ...
+              <h2>Lorem Ipsum Dolor sit amet</h2>
```
Comment (lycoangelo) [inline view/browse-page.html]: Then let's change this to `h3` since we are updating `Continue exploring below` to become an `h3`.

**Review by lycoangelo — CHANGES_REQUESTED** [review]

**Comment by nicoRDL** [comment]
@lycoangelo added all change requests and code improvements, for review

b3b52f20dea348a5137e38d5fb35993fb0ed8a29

File: view/index.html [hunk 10]
```diff
@@ -1,104 +1,158 @@
...
-              <p class="author-description"> 
-                Author is the primary contributor to the site, both content and structure. Author is growing enthusiast for custom builds and modding.
-              </p>
-            </div>
-        </section>
+              <a href="detail-page-sample3.html" class="article-card-expand-btn">continue reading ...</a>
+          </div>
+        </div>
+        <button class="articles-section-expand-btn" href="browse-page.html">
+            See More Articles
```
Comment (lycoangelo) [inline view/index.html]: @nicoRDL, the indentations here are still off.

File: view/browse-page.html [hunk 11]
```diff
@@ -79,182 +79,184 @@ <h2>Typing adjustment and how to make it easier</h2>
...
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
-            eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
-          <button class="article-card-expand-btn">
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-light-sample1.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/browse-page.html]: The indentations here are still off.

File: view/browse-page.html [hunk 12]
```diff
@@ -79,182 +79,184 @@ <h2>Typing adjustment and how to make it easier</h2>
...
-              continue reading ...
+              <h2>Lorem Ipsum Dolor sit amet</h2>
             </a>
-          </button>
-        </figure>
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-light-sample2.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/browse-page.html]: The indentations here are still off.

File: view/index.html [hunk 13]
```diff
@@ -1,104 +1,158 @@
...
+          />
+          <div class="recent-post-body">
+            <a href="detail-page-sample1.html">
+              <h2>Down-sizing your keyboard: should you even?</h2>
+            </a>
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-sample1.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/index.html]: The indentations here are still off.

File: view/index.html [hunk 14]
```diff
@@ -1,104 +1,158 @@
...
+          />
+          <div class="recent-post-body">
+            <a href="detail-page-sample2.html">
+              <h2>How to start a Clean Setup. Coffee not included.</h2>
+            </a>
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-sample2.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/index.html]: The indentations here are still off.

File: view/index.html [hunk 15]
```diff
@@ -1,104 +1,158 @@
...
-              <img src="../images/user.png" alt="author photo">  
-              <h2 class="author-name">
-                Main Author
-              </h2>
-              <p class="author-description"> 
-                Author is the primary contributor to the site, both content and structure. Author is growing enthusiast for custom builds and modding.
-              </p>
-            </div>
-        </section>
+              <a href="detail-page-sample3.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/index.html]: The indentations here are still off.

File: view/browse-page.html [hunk 16]
```diff
@@ -79,182 +79,184 @@ <h2>Typing adjustment and how to make it easier</h2>
...
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
-            eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
-          <button class="article-card-expand-btn">
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-light-sample1.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/browse-page.html]: The indentations here are still off.

File: view/browse-page.html [hunk 17]
```diff
@@ -79,182 +79,184 @@ <h2>Typing adjustment and how to make it easier</h2>
...
-            eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
-          <button class="article-card-expand-btn">
-            <a href="detail-page-light-sample2.html">
-              continue reading ...
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-light-sample2.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/browse-page.html]: The indentations here are still off.

File: view/browse-page.html [hunk 18]
```diff
@@ -79,182 +79,184 @@ <h2>Typing adjustment and how to make it easier</h2>
...
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
-            eiusmod tempor incididunt ut labore et dolore magna aliqua.
-          </p>
-          <button class="article-card-expand-btn">
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+              eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            </p>
+              <a href="detail-page-light-sample1.html" class="article-card-expand-btn">continue reading ...</a>
```
Comment (lycoangelo) [inline view/browse-page.html]: The indentations here are still off.

**Review by lycoangelo — CHANGES_REQUESTED** [review]
@nicoRDL,  have left another round of comments pointing out inconsistent indentations.

**Comment by nicoRDL** [comment]
@lycoangelo noted on the spacing and indention issues. Re-applied the formatter on the target pages, hopefully it fixes all the errors. For your re-review

https://github.com/nicoRDL/dotTheory-FEv2/pull/5/commits/415e18d7a2091249ec65b5017f5559c2fc529abf

**Pull request merged**